### PR TITLE
Implement a total number of nodes and pods quota for k8s.

### PR DESCRIPTION
**Pull request by scrosby**
## Changes proposed in this PR

- Implement a total number of nodes and pods quota for k8s.

## Why are we making these changes?

So we can control the load we put on k8s.

### CODE REVIEW

File: scheduler/src/cook/kubernetes/compute_cluster.clj [hunk 1]
```diff
@@ -345,64 +345,72 @@
                                             (add-starting-pods this)
                                             (filter synthetic-pod->job-uuid))
             num-synthetic-pods (count outstanding-synthetic-pods)
-            {:keys [image user command max-pods-outstanding] :or {command "exit 0"}} synthetic-pods-config]
+            total-pods (-> @all-pods-atom keys count)
+            total-nodes (-> @current-nodes-atom keys count)
+            {:keys [image user command max-pods-outstanding max-total-pods max-total-nodes]
+             :or {command "exit 0" max-total-pods 1000000 max-total-nodes 1000000}} synthetic-pods-config]
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/compute_cluster.clj:351]: Why not have much lower defaults, maybe 32000 total pods and 1000 total nodes?

File: scheduler/src/cook/kubernetes/compute_cluster.clj [hunk 2]
```diff
@@ -345,64 +345,72 @@
...
-            (let [timer-context-launch-tasks (timers/start (metrics/timer "cc-synthetic-pod-launch-tasks" name))]
-              (cc/launch-tasks this
-                               synthetic-task-pool-name
-                               [{:task-metadata-seq task-metadata-seq}]
-                               (fn [_]))
-              (.stop timer-context-launch-tasks))))
+        (let [max-launchable (min (- max-pods-outstanding num-synthetic-pods)
+                                  (- max-total-nodes total-nodes)
+                                  (- max-total-pods total-pods))]
+          (if (not (pos? max-launchable))
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/compute_cluster.clj:359]: Can we add a unit test for this?

Comment (scrosby) [reply scheduler/src/cook/kubernetes/compute_cluster.clj:359]: Done.

File: scheduler/src/cook/kubernetes/compute_cluster.clj (hunk 1 above)
Comment (scrosby) [reply scheduler/src/cook/kubernetes/compute_cluster.clj:351]: Ok.

File: scheduler/src/cook/test/testutil.clj [hunk 3]
```diff
@@ -543,11 +543,11 @@
...
-  [namespaced-pod-name->pod pool-names synthetic-pods-user node-blocklist-labels]
-  (let [synthetic-pods-config {:image "image"
-                               :user synthetic-pods-user
-                               :max-pods-outstanding 4
-                               :pools pool-names}]
+  [namespaced-pod-name->pod pool-names node-blocklist-labels additional-synthetic-pods-config]
+  (let [synthetic-pods-config (merge {:image "image"
+                                      :user nil
+                                      :max-pods-outstanding 4
+                                      :pools pool-names} additional-synthetic-pods-config)]
```
Comment (dposada) [inline scheduler/src/cook/test/testutil.clj:550]:
```suggestion
  (let [synthetic-pods-config (merge {:image "image"
                                      :user nil
                                      :max-pods-outstanding 4
                                      :pools pool-names} 
                                     additional-synthetic-pods-config)]
```

**Pull request merged**